soup.go

```diff
@@ -133,14 +132,44 @@ func (r Root) Find(args ...string) Root {
 // and returns an array of structs, each having
 // the respective pointers
 func (r Root) FindAll(args ...string) []Root {
-	temp := findAllofem(r.Pointer, args)
+	temp := findAllofem(r.Pointer, args, false)
 	if len(temp) == 0 {
 		if debug {
 			panic("Element `" + args[0] + "` with attributes `" + strings.Join(args[1:], " ") + "` not found")
 		}
 		return []Root{}
 	}
-	pointers := make([]Root, 0, 10)
+	pointers := make([]Root, 0, len(temp))
+	for i := 0; i < len(temp); i++ {
+		pointers = append(pointers, Root{temp[i], temp[i].Data, nil})
+	}
+	return pointers
+}
+
+// FindStrict finds the first occurrence of the given tag name
+// only if all provided attribute's values are exists
+func (r Root) FindStrict(args ...string) Root {
+	temp, ok := findOnce(r.Pointer, args, false, true)
+	if ok == false {
+		if debug {
+			panic("Element `" + args[0] + "` with attributes `" + strings.Join(args[1:], " ") + "` not found")
+		}
+		return Root{nil, "", errors.New("element `" + args[0] + "` with attributes `" + strings.Join(args[1:], " ") + "` not found")}
+	}
+	return Root{temp, temp.Data, nil}
+}
+
+// FindAllStrict finds all occurrences of the given tag name
+// only if all provided attribute's values are exists
+func (r Root) FindAllStrict(args ...string) []Root {
```

Comment:
Again, `// only if all the values of the provided attribute are an exact match` would be a better grammatical fit